### PR TITLE
Remove spring binder test module

### PR DESCRIPTION
**Pull request by moarychan**
This PR is used to enable the [new feature](https://github.com/Azure/azure-sdk-for-java/pull/26002) of the pipeline for supporting multiple spring boot and spring cloud version compatibilities.

Due to the jackson version conflicts, we finally decide to use the single module to run our integration test cases. WIll move to the corresponding artifact id module in future.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-test-eventhubs-binder/src/test/java/com/azure/spring/sample/eventhubs/binder/EventHubsBinderSyncModeIT.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-package com.azure.spring.cloud.stream.binder.eventhubs;
+package com.azure.spring.sample.eventhubs.binder;
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-test-eventhubs-binder/src/test/java/com/azure/spring/sample/eventhubs/binder/EventHubsBinderSyncModeIT.java:4]: nit: I think we should not use the package sample here.

**Review by saragluna — APPROVED** [review]
LGTM.

File: sdk/spring/spring-cloud-azure-test-eventhubs-binder/src/test/java/com/azure/spring/sample/eventhubs/binder/EventHubsBinderSyncModeIT.java (hunk 1 above)
Comment (moarychan) [reply sdk/spring/spring-cloud-azure-test-eventhubs-binder/src/test/java/com/azure/spring/sample/eventhubs/binder/EventHubsBinderSyncModeIT.java:4]: I will change it in the next PR.

**Pull request merged**